staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/object_count_tracker.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package request
+
+import (
+	"sync"
+)
+
+// StorageObjectCountTracker is an interface that is used to keep track of
+// of the total number of storage objects for each resource.
+// {group}.{resource} is used as the key name to update and retrieve
+// the total number of objects for a given resource.
+type StorageObjectCountTracker interface {
+	// OnCount is invoked to update the current number of total
+	// objects for the given resource
+	OnCount(string, int64)
+
+	// Get returns the total number of objects for the given resource
+	Get(string) (int64, bool)
+}
+
+func NewStorageObjectCountTracker() StorageObjectCountTracker {
+	tracker := &objectCountTracker{
+		counts: map[string]int64{},
+	}
+	return tracker
+}
+
+// it implements StorageObjectCountTracker with reader/writer mutual exclusion lock.
+// TODO: since this falls in the path of every incoming request we can
```

Comment:
Actually, i don't believe any actual requests for Count originate from outside the apiserver. IIRC, all count invocations occur in some async goroutine at regular intervals, specifically for the purpose of updating the apiserver_storage_object_count metric.